distributed/core.py

```diff
@@ -525,17 +514,22 @@ def start_periodic_callbacks(self):
                 pc.start()
 
     def stop(self):
-        if not self.__stopped:
-            self.__stopped = True
+        if self.__stopped:
+            return
 
-            for listener in self.listeners:
+        self.__stopped = True
+        _stops = set()
+        for listener in self.listeners:
+            future = listener.stop()
+            if inspect.isawaitable(future):
+                _stops.add(future)
```

Comment:
IIUC you introduce this merely to ensure that there are no users outside that provide a listener with an async stop?
This smells like we should introduce a deprecation warning and get rid of this